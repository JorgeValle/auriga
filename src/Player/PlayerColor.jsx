import React from 'react';

// Components should represent one piece of data model
class PlayerColor extends React.Component {
  render() {

    const player = this.props.player;

    return (
      <div className="player-color">
        <span className={player.color}>{player.color}</span>
      </div>
    );
  }
}

export default PlayerColor;