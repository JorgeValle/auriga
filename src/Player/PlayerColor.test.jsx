import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import PlayerColor from './PlayerColor.jsx';

configure({
  adapter: new Adapter()
});

describe('<PlayerColor />', () => {

  // We shallow render to make sure changes in children don't cause cascading fails
  it('shallow renders', () => {
    const wrapper = shallow(<PlayerColor />);
  });

  // Snapshot matching
  it('matches snapshot', () => {
    const wrapper = shallow(<PlayerColor />);
    expect(wrapper).toMatchSnapshot();
  });

});

