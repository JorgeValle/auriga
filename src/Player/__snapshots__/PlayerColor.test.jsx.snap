// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlayerColor /> matches snapshot 1`] = `ShallowWrapper {}`;
